core/src/main/kotlin/no/ntnu/beardblaster/screen/GameplayScreen.kt

```diff
@@ -1,105 +1,76 @@
+    batch: Batch,
+    assets: AssetStorage,
+    camera: OrthographicCamera,
+) : BaseScreen(game, batch, assets, camera) {
+    private val attackBtn = scene2d.textButton(Nls.attack())
+    private val quiteBtn = scene2d.textButton(Nls.quit())
+
+    private val stage: Stage by lazy {
         val result = BeardBlasterStage()
         Gdx.input.inputProcessor = result
         result
     }
 
     override fun show() {
-        LOG.debug { "GAMEPLAY Screen" }
-
-        skin = Skin(Assets.assetManager.get(Assets.atlas))
-        table = Table(skin)
-        table.setBounds(0f, 0f, viewport.worldWidth, viewport.worldHeight)
-
-        // Fonts
-        val standardFont = Assets.assetManager.get(Assets.standardFont)
-
-        // Creating buttons
-        val textButtonStyle = TextButton.TextButtonStyle()
-        skin.getDrawable("button_default").also { textButtonStyle.up = it }
-        skin.getDrawable("button_default_hover").also { textButtonStyle.over = it }
-        skin.getDrawable("button_default_pressed").also { textButtonStyle.down = it }
-        textButtonStyle.pressedOffsetX = 4f
-        textButtonStyle.pressedOffsetY = -4f
-        textButtonStyle.font = standardFont
-
-        btnAttack = TextButton("ATTACK", textButtonStyle)
-        btnQuit = TextButton("QUIT", textButtonStyle)
         setBtnEventListeners()
-
-        Label.LabelStyle(standardFont, Color.BLACK).also {
-            heading = Label("Preparation phase", it)
-            heading.setFontScale(2f)
-            it.background = skin.getDrawable("modal_fancy_header")
-            heading.setAlignment(Align.center)
+        val table = scene2d.table {
+            setBounds(0f, 0f, WIDTH, HEIGHT)
+            add(headingLabel(Nls.preparationPhase())).pad(50f)
+            row()
+            add(attackBtn).pad(40f)
+            row()
+            add(quiteBtn).pad(40f)
+            row()
         }
-
-        // Creating table
-        table.add(heading).pad(50f)
-        table.row()
-        table.add(btnAttack).pad(40f)
-        table.row()
-        table.add(btnQuit).pad(40f)
-        table.row()
-
-        // Adding actors to the stage
-        gameplayStage.addActor(table)
-
-        Gdx.input.inputProcessor = gameplayStage
-
-    }
-
-    override fun update(delta: Float) {
+        stage.clear()
+        stage.addActor(table)
+        Gdx.input.inputProcessor = stage
     }
 
     override fun setBtnEventListeners() {
-        btnAttack.onClick {
-            LOG.debug { "Wizard 1 attacks" }
+        attackBtn.onClick {
+            log.info { Nls.wizardNAttacks(1) }
         }
-        btnQuit.onClick {
+        quiteBtn.onClick {
             game.removeScreen<GameplayScreen>()
-            game.addScreen(GameplayScreen(game))
+            // FIXME Really needed to add a new game play screen here?
+            // Might be better to add it when joining/starting a new game.
+            game.addScreen(GameplayScreen(game, batch, assets, camera))
```

Comment:
Agree with this thought. We can move adding the gameplay screen to LobbyScreen

Reply:
Tried to move `addScreen` within `startButton.OnClick` in the `LobbyScreen`, but that resulted in:

```
com.badlogic.gdx.utils.GdxRuntimeException: Screen already registered to type: class no.ntnu.beardblaster.screen.GameplayScreen.
```

Leaving it as it is, until we figure out how to handle it.